app/routes/resource/get-blog-post.$slug.ts

```diff
@@ -0,0 +1,20 @@
+import { type LoaderArgs } from "@remix-run/server-runtime";
+import { getComponents } from "blog/getComponents";
+import { parseMdx } from "blog/parser";
+import { POSTS_BUILD_DIR } from "blog/paths";
+
+export const loader = async function ({ request, params }: LoaderArgs) {
+  const { slug = "" } = params;
+  const url = new URL(`${new URL(request.url).origin}/posts/${slug}.mdx`);
```

Comment:
now this url will always be relative right? and thus you can do ``new URL(`/posts/${slug}.mdx`)`` right?

it also might be marginally better to check the slug w/ an invariant before making the fetch request

Reply:
Unfortunately, using a relative URL isn't an option. It looks like Node has a different version of `URL` from the browser where it requires an absolute URL. Otherwise it throws. Same for `fetch` too. That's annoying.

But I did use `invariant` for the slug. So we got that piece.